lib/travis/listener/app.rb

```diff
@@ -159,7 +186,15 @@ def delivery_guid
       end
 
       def payload
-        params[:payload]
+        if github_pr_event?
+          params[:payload]
+        elsif github_apps_event?
+          begin
+            @_parsed_json ||= JSON.parse(request.body.read)
+          rescue JSON::ParserError
+            nil
+          end
+        end
```

Comment:
@joshk and I noticed that the structure of the existing webhooks (form data) differs from Apps (which sends JSON) thus this bit of switching